src/async_impl/request.rs

```diff
@@ -570,6 +570,31 @@ where
     }
 }
 
+impl TryFrom<Request> for HttpRequest<Option<Body>> {
```

Comment:
Oh hm, the `Option` for a body is a tricky one. That's the question, when turning into an `http::Request`, should we point out the body might have been absent (`None`), or just fill in an "empty" body? hyper does the latter, normally...

Reply:
I didn't look at the surrounding code much last time. Have now found `Body::empty` and will fall back to that. I'm still not sure whether to make this impl generic. I'm leaning towards no since most interesting conversions couldn't be done using `Into` like the other way around and `http::Request` has `map` and `into_parts` / `from_parts` which can be used for simple (infallible, non-async) and more complex body conversions respectively.